lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -82,7 +85,7 @@ unsigned swift::getObjCDiagnosticAttrKind(ObjCReason reason) {
   case ObjCReason::MemberOfObjCMembersClass:
   case ObjCReason::ElementOfObjCEnum:
   case ObjCReason::Accessor:
-    llvm_unreachable("should not diagnose this @objc reason");
+    return ~static_cast<unsigned>(0);
```

Comment:
Could you clarify why this is doing `~static_cast<unsigned>(0)`, instead of (say) `static_cast<unsigned>(reason);`?

Is this code path unreachable right now but it ends up being reachable with the changes for access notes?

Reply:
Right—this is no longer unreachable, but any diagnostic which would use one of these cases should be ignored. I can add a comment to that effect.

Reply:
```
    // Diagnostics involving these ObjCReasons should always be ignored, so we
    // deliberately return a value which is out of bounds for the `%select` in
    // `OBJC_ATTR_SELECT`.
    return ~static_cast<unsigned>(0);
```